src/main/java/org/jenkinsci/remoting/engine/JnlpAgentEndpointResolver.java

```diff
@@ -349,29 +350,15 @@ public JnlpAgentEndpoint resolve() throws IOException {
 
     @SuppressFBWarnings(value = "UNENCRYPTED_SOCKET", justification = "This just verifies connection to the port. No data is transmitted.")
     private boolean isPortVisible(String hostname, int port) {
-        boolean exitStatus = false;
-        Socket s = null;
-
-        try {
-            s = new Socket();
+        try (Socket s = new Socket()) {
             s.setReuseAddress(true);
             SocketAddress sa = new InetSocketAddress(hostname, port);
             s.connect(sa, 5000);
+            return s.isConnected();
         } catch (IOException e) {
             LOGGER.warning(e.getMessage());
-        } finally {
-            if (s != null) {
-                if (s.isConnected()) {
-                    exitStatus = true;
```

Comment:
Change of the behavior. when IOException happens. New code will always return false in this case